package.json

```
@@ -11,7 +11,7 @@
     "build": "npm run build-shared && npm run build-frontend && npm run build-backend",
     "server": "npm run dev --prefix server",
     "client": "npm start --prefix client",
-    "dev": "concurrently \"./wait-for-it.sh localhost:3306 -t 0 -- npm run build-shared && sleep 5 && npm run server\" \"npm run client\" \"docker-compose up \"",
+    "dev": "concurrently \"./wait-for-it.sh localhost:3306 -t 0 -- npm run build-shared && npm run build-frontend && sleep 5 && npm run server\" \"npm run client\" \"docker-compose up \"",
```

```suggestion
    "dev": "concurrently \"./wait-for-it.sh localhost:3306 -t 0 -- npm run build-shared && sleep 5 && npm run server\" \"npm run client\" \"docker-compose up \"",
```
Building the frontend isn't actually needed, since that's covered by `npm run client`

---

I thought it would be too.

`npm run client` does build the client, but it doesn't write the files to `client/build`, and even though the frontend hosted by the `npm run client` server will be used, the backend will fail to start if `client/build/index.html` doesn't exist. If you delete that directory and then do `npm run client` in one terminal, wait for it to finish, and then do `npm run server` in another terminal, you will get an error about not being able to find the client files.

Alternatively, if `server/src/bootstrap/index.ts` somehow knew it was running in a development setup it could skip trying to read `index.html`.

---

Thanks for bringing this up - I'll be doing the latter soon 👍